### PR TITLE
Yuval/hypervisor oriented

### PR DESCRIPTION
**Pull request by yuval-lb**
1. refactor makefile not build osmosis everytime - save a lot of time in inaugurator creation
2. adding --hypervisor flag to make root partition big as possible (dont waste on osmosis/etcd)

### CODE REVIEW

**Comment by yuval-lb** [comment]
verified with different machine labels

File: Makefile.build [hunk 1]
```diff
@@ -67,9 +67,10 @@ build/inaugurator.%.initrd.img: $(INITRD_IMAGE_PATH)
 	mv $@.tmp $@
 
 build/osmosis-1.0.linux-x86_64.tar.gz:
-	(cd ../osmosis/ && $(MAKE) clean && $(MAKE) build && $(MAKE) egg)
-	 cp ../osmosis/dist/osmosis-1.0.linux-x86_64.tar.gz $@
-
+	if  test ! -f ../osmosis/build/cpp/osmosis.bin ; then\
```
Comment (abel-lb) [inline Makefile.build]: but how you know the file version matches the source code ? shouldn't you actually check if it's in workspace/build/[dev/rel]/osmosis with matching version and take it from there.. if not build ?

File: inaugurator/partitiontable.py [hunk 2]
```diff
@@ -22,9 +22,12 @@ class PartitionTable:
                                                    lvm=dict(set_flags="lvm", flags="lvm", sizeMB="fillUp")),
                                    order=("boot", "lvm")))
 
-    def __init__(self, device, sizesGB=dict(), layoutScheme="GPT"):
+    def __init__(self, device, sizesGB=dict(), layoutScheme="GPT", hypervisor=False):
         self._sizesGB = dict(self._DEFAULT_SIZES_GB)
         self._sizesGB.update(sizesGB)
+        self._hypervisor = hypervisor
+        if hypervisor:
+            self._set_hyperviosr()
```
Comment (abel-lb) [inline inaugurator/partitiontable.py]: shouldn't you actually handle the case is not hypervisor and reset volume groups ? 
I mean what will happen if we:
(1) allocate a machine that was used as hypervisor now as a client (non-hypervisors) or other
(2) allocate a machine that was used as a client (non-hypervisors) as an hypervisor ?

File: Makefile.build (hunk 1 above)
Comment (yuval-lb) [reply Makefile.build]: yes, I will add it as TBD

File: inaugurator/partitiontable.py (hunk 2 above)
Comment (yuval-lb) [reply inaugurator/partitiontable.py]: those cases are handled - "_set_hypervisor" changes the VOLUME_GROUP name. once this string is changed the inaugurator knows to recreate the volumes.

**Pull request merged**